app/models/event.rb

```diff
@@ -0,0 +1,21 @@
+class Event < ApplicationRecord
+  belongs_to :user, optional: true
+  belongs_to :application, optional: true
+  belongs_to :course, optional: true
+  belongs_to :lead_provider, optional: true
+  belongs_to :school, optional: true
+  belongs_to :statement, optional: true
+  belongs_to :statement_item, optional: true
+  belongs_to :declaration, optional: true
```

Comment:
With all these being optional it would probably make sense to have a validation rule that checks at least one of them is present.